File: Perhotelan_VB/Perhotelan/f_riwayat_tamu.vb
Imports System.Data.Odbc

Public Class f_riwayat_tamu

    Dim conn As OdbcConnection
    Dim cmd As OdbcCommand
    Dim dr As OdbcDataReader
    Dim da As OdbcDataAdapter
    Dim dt As DataTable
    Dim ds As DataSet

    Sub koneksi()
        Try
            conn = New OdbcConnection
            conn.ConnectionString = "dsn=db_hotel"
            conn.Open()
        Catch ex As Exception
            MsgBox("Koneksi Ke Database Gagal", vbInformation, "Pemberitahuan")
        End Try
    End Sub

    Sub tampil_riwayat_tamu()
        Try
            koneksi()
            Dgv_tamu.Rows.Clear()
            Dim sql As String = "select * from tb_riwayat_tamu"
            da = New OdbcDataAdapter(sql, conn)
            dt = New DataTable
            da.Fill(dt)
            For Each row In dt.Rows
                Dgv_tamu.Rows.Add(row("id"), row("nama_tamu"), row("jenis_kelamin"),
                                   row("alamat"), row("no_telp"), row("no_kamar"), row("durasi"),
                                   row("cek_in"), row("cek_out"))
            Next
            dt.Rows.Clear()
        Catch ex As Exception
            MsgBox("Gagal Menampilkan Data", vbInformation, "Pemberitahuan")
        End Try
    End Sub
    Private Sub f_riwayat_tamu_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        tampil_riwayat_tamu()
    End Sub

    Private Sub B_cari_Click(sender As Object, e As EventArgs) Handles B_cari.Click
        If T_cari.Text = "" Then
            MsgBox("Kolom Pencarian Masih Kosong", vbInformation, "Pemberitahuan")
        Else
            koneksi()
            Dgv_tamu.Rows.Clear()
            Try
                Dim sql As String = "select * from tb_riwayat_tamu where " +
                                    "id like '%" + T_cari.Text + "%'" +
                                    " OR nama_tamu like '%" + T_cari.Text + "%'" +
                                    " OR jenis_kelamin like '%" + T_cari.Text + "%'" +
                                    " OR alamat like '%" + T_cari.Text + "%'" +
                                    " OR no_telp like '%" + T_cari.Text + "%'" +
                                    " OR no_kamar like '%" + T_cari.Text + "%'" +
                                    " OR lantai like '%" + T_cari.Text + "%'" +
                                    " OR tipe_kamar like '%" + T_cari.Text + "%'" +
                                    " OR durasi like '%" + T_cari.Text + "%'" +
                                    " OR total_biaya like '%" + T_cari.Text + "%'" +
                                    " OR cek_in like '%" + T_cari.Text + "%'" +
                                    " OR cek_out like '%" + T_cari.Text + "%'" +
                                    " order by id asc"
                da = New OdbcDataAdapter(sql, conn)
                dt = New DataTable
                da.Fill(dt)
                For Each row In dt.Rows
                    Dgv_tamu.Rows.Add(row("id"), row("nama_tamu"), row("jenis_kelamin"),
                                   row("alamat"), row("no_telp"), row("no_kamar"), row("durasi"),
                                   row("cek_in"), row("cek_out"))
                Next
                dt.Rows.Clear()
            Catch ex As Exception
                MsgBox("Data Tidak Ditemukan", vbInformation, "Pemberitahuan")
            End Try
        End If
    End Sub

    Private Sub T_cari_KeyDown(sender As Object, e As KeyEventArgs) Handles T_cari.KeyDown
        If e.KeyCode = Keys.Enter Then
            B_cari_Click(sender, e)
        End If
    End Sub

    Private Sub B_refresh_Click_1(sender As Object, e As EventArgs) Handles B_refresh.Click
        tampil_riwayat_tamu()
    End Sub

    Private Sub f_riwayat_tamu_Closed(sender As Object, e As EventArgs) Handles Me.Closed
        f_tamu.Visible = True
    End Sub

End Class

File: Perhotelan_C#/Perhotelan/f_riwayat_tamu.cs
using System;
using System.Data;
using System.Data.Odbc;
using System.Windows.Forms;
using Microsoft.VisualBasic;

namespace Perhotelan
{
    public partial class f_riwayat_tamu
    {
        public f_riwayat_tamu()
        {
            InitializeComponent();
            _Panel1.Name = "Panel1";
            _Label1.Name = "Label1";
            _Dgv_tamu.Name = "Dgv_tamu";
            _T_cari.Name = "T_cari";
            _B_cari.Name = "B_cari";
            _Column1.Name = "Column1";
            _Column2.Name = "Column2";
            _Column3.Name = "Column3";
            _Column4.Name = "Column4";
            _Column5.Name = "Column5";
            _Column6.Name = "Column6";
            _Column7.Name = "Column7";
            _Column8.Name = "Column8";
            _Column9.Name = "Column9";
            _Column10.Name = "Column10";
            _Column11.Name = "Column11";
            _Column12.Name = "Column12";
            _Column13.Name = "Column13";
            _B_refresh.Name = "B_refresh";
        }

        private OdbcConnection conn;
        private OdbcCommand cmd;
        private OdbcDataReader dr;
        private OdbcDataAdapter da;
        private DataTable dt;
        private DataSet ds;

        public void koneksi()
        {
            try
            {
                conn = new OdbcConnection();
                conn.ConnectionString = "dsn=db_hotel";
                conn.Open();
            }
            catch (Exception ex)
            {
                Interaction.MsgBox("Koneksi Ke Database Gagal", Constants.vbInformation, "Pemberitahuan");
            }
        }

        public void tampil_riwayat_tamu()
        {
            try
            {
                koneksi();
                Dgv_tamu.Rows.Clear();
                string sql = "select * from tb_riwayat_tamu";
                da = new OdbcDataAdapter(sql, conn);
                dt = new DataTable();
                da.Fill(dt);
                foreach (DataRow row in dt.Rows)
                {
                    Dgv_tamu.Rows.Add(row.ItemArray);
                }
                dt.Rows.Clear();
            }
            catch (Exception ex)
            {
                Interaction.MsgBox("Gagal Menampilkan Data", Constants.vbInformation, "Pemberitahuan");
            }
        }

        private void f_riwayat_tamu_Load(object sender, EventArgs e)
        {
            tampil_riwayat_tamu();
        }

        private void B_cari_Click(object sender, EventArgs e)
        {
            if (string.IsNullOrEmpty(T_cari.Text))
            {
                Interaction.MsgBox("Kolom Pencarian Masih Kosong", Constants.vbInformation, "Pemberitahuan");
            }
            else
            {
                koneksi();
                Dgv_tamu.Rows.Clear();
                try
                {
                    string sql = "select * from tb_riwayat_tamu where " 
                                 + "id like '%" + T_cari.Text + "%'" 
                                 + " OR nama_tamu like '%" + T_cari.Text + "%'" 
                                 + " OR jenis_kelamin like '%" + T_cari.Text + "%'" 
                                 + " OR alamat like '%" + T_cari.Text + "%'" 
                                 + " OR no_telp like '%" + T_cari.Text + "%'" 
                                 + " OR no_kamar like '%" + T_cari.Text + "%'" 
                                 + " OR lantai like '%" + T_cari.Text + "%'" 
                                 + " OR tipe_kamar like '%" + T_cari.Text + "%'" 
                                 + " OR durasi like '%" + T_cari.Text + "%'" 
                                 + " OR total_biaya like '%" + T_cari.Text + "%'"
                                 + " OR cek_in like '%" + T_cari.Text + "%'" 
                                 + " OR cek_out like '%" + T_cari.Text + "%'" 
                                 + " order by id asc";
                    da = new OdbcDataAdapter(sql, conn);
                    dt = new DataTable();
                    da.Fill(dt);
                    foreach (DataRow row in dt.Rows)
                    {
                        Dgv_tamu.Rows.Add(row.ItemArray);
                    }
                    dt.Rows.Clear();
                }
                catch (Exception ex)
                {
                    Interaction.MsgBox("Data Tidak Ditemukan", Constants.vbInformation, "Pemberitahuan");
                }
            }
        }

        private void T_cari_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Enter)
            {
                B_cari_Click(sender, e);
            }
        }

        private void B_refresh_Click_1(object sender, EventArgs e)
        {
            tampil_riwayat_tamu();
        }

        private void f_riwayat_tamu_Closed(object sender, EventArgs e)
        {
            My.MyProject.Forms.f_tamu.Visible = true;
        }
    }
}